docs/iam-usage.rst

```diff
@@ -0,0 +1,148 @@
+False
+
+Same as above, but removes the specified roles from the specified member.
+
+>>> resource.add_members(iam.roles.OWNER.name, [iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com')])
+True
+
+The first argument is a role, and the second argument is a list of members to be added to that role.
+
+>>> resource.remove_members(iam.roles.OWNER.name, iam.ALL_USERS)
+True
+
+Same as above, but removes all listed members from the specified role.
+
+>>> def remove_fn(member):
+>>>     return iam.is_group(member) or member == iam.user('bob@example.com')
+>>> policy_change = iam.PolicyChange(version=2).add(iam.roles.OWNER.name, [iam.user('alice@example.com')])
+>>> policy_change.remove(iam.roles.EDITOR.name, [iam.domain('example.com'), iam.group('devs@example.com')])
+>>>              .remove_fn(iam.roles.READER.name, remove_fn)
+>>> policy_change.apply(resource)
+True
+
+If you need more complex logic to modify an IAM policy you can create a ``iam.PolicyChange`` object. ``PolicyChange`` takes
+an optional ``version`` keyword argument which specifies which version the updated policy should be considered.
+
+``iam.PolicyChange`` exposes four methods. ``add`` and ``remove`` take a role, and a list of members, and add, or remove
+those members from the specified roll respectively. They have the same signature as ``resource.add_members`` and ``resource.remove_members`` respectively.
+``remove_fn`` takes a role and a function object as arguments.
+The provided function takes a ``member`` string and returns whether or not the member should have the specified role.
+
+Finally, ``iam.PolicyChange`` exposes an ``apply`` method, which takes a resource, and applies the change to the resource.
+
+Just like above, this method returns True if the change was successfully made, or False otherwise.
+
+>>> resource.set_policy(
+>>>     {
+>>>         iam.roles.OWNER.name: [iam.user('alice@example.com')],
+>>>         iam.roles.EDITOR.name: [(iam.user('alice@example.com'), (iam.user('charles@example.com')]
+>>>     },
+>>>     etag=XXXXXX,
+>>>     version=0
+>>>)
+
+Finally, you can manually set the policy of a resource.
+``set_policy`` takes a policy, and optionally an ``etag`` string, and version number.
+
+Use this only if you don't need any transactionality guarantees, or want to handle transactionality yourself, using etag.
+
+If updates are made to your policy during this change, they will be overwritten with exactly what is in your policy.
+Or, if an etag is specified they will fail with a ``TransactionInterruptedException``
+
+
+Policy Access Methods
+---------------------
+
+As seen above, IAM Policies are simply dictionaries, of the following form.
+
+``{role_string: [member_string, member_string], role_string...}``
+
+>>> resource.get_policy()
+{'roles/owner': ['user:alice@example.com')], 'roles/editor': [('user:alice@example.com'), ('user:charles@example.com')]}, 0, XXXXX
```

Comment:
Ditto here on readability. Also, split the 3 return values out and print each individually.